src/services/catalog.service.ts

```diff
@@ -2,16 +2,70 @@ import * as angular from 'angular';
 import * as _ from 'lodash';
 
 export class CatalogService {
-  static $inject = ['Constants', '$filter'];
+  static $inject = ['$filter', '$q', 'Constants', 'DataService', 'Logger'];
 
   public $filter: any;
-  private constants: any;
+  public catalogServiceItems: any;
+
+  private $q: any;
   private categories: any;
+  private constants: any;
+  private dataService: any;
+  private logger: any;
 
-  constructor(constants: any, $filter: any) {
-    this.constants = constants;
+  constructor($filter: any, $q: any, constants: any, DataService: any, Logger: any) {
     this.$filter = $filter;
+    this.$q = $q;
+    this.constants = constants;
     this.categories = this.constants.SERVICE_CATALOG_CATEGORIES;
+    this.dataService = DataService;
+    this.logger = Logger;
+  }
+
+  public getCatalogItems() {
+    let deferred: any = this.$q.defer();
+    let serviceClasses: any, imageStreams: any;
+
+    let promises = {
+      serviceClasses: this.dataService.list({
+        group: 'servicecatalog.k8s.io',
+        resource: 'serviceclasses'
+      }, {}),
+      imageStreams: this.dataService.list("imagestreams", {namespace: "openshift"})
+    };
+
+    this.$q.all(promises).then((values) => {
```

Comment:
I tried:
```
public getCatalogItems() {
    return this.$q.all({
      serviceClasses: this.dataService.list({
        group: 'servicecatalog.k8s.io',
        resource: 'serviceclasses'
      }, {}),
      imageStreams: this.dataService.list("imagestreams", {namespace: "openshift"})
    }).then((values) => {
      let serviceClasses: any = values.serviceClasses.by("metadata.name");
      let imageStreams: any = values.imageStreams.by("metadata.name");
      this.catalogServiceItems = this.convertToServiceItems(serviceClasses, imageStreams);
      this.$q.resolve(this.catalogServiceItems);
    }, () => {
      this.logger.log("Error Loading Catalog Items");
      this.$q.reject();
    });
  }
```
But that causes the an endless spinner in the service-view, and no 'Recently Viewed' in Project Summary.